pkg/controllers/scheduling/schedule.go

```diff
@@ -261,11 +269,39 @@ func mergeWeights(defaultWeight map[string]int32, customizedWeight []clusterapiv
 	}
 	// override the default weight
 	for _, c := range customizedWeight {
-		weights[c.Name] = c.Weight
+		if c.ScoreCoordinate == nil {
+			weights[c.Name] = c.Weight
```

Comment:
I think you can use coordinate struct as the key for weight.

Reply:
fixed.